runtime/src/main/cpp/CyclicCollector.cpp

```diff
@@ -361,6 +361,11 @@ class CyclicCollector {
     CHECK_CALL(pthread_cond_signal(&cond_), "Cannot signal collector")
   }
 
+  void localGC() {
+    // We just need to take GC lock here, to avoid release of object we walk on.
+    Locker locker(&lock_);
```

Comment:
This and other cases of taking GC lock could block the UI thread for long.
Isn't it enough to notify GC with a flag like in `mutateRoot`?

Reply:
Maybe, but first I want to be provably correct.

Reply:
Added TODO.